packages/ckeditor5-link/src/linkui.ts

```diff
@@ -487,30 +552,43 @@ export default class LinkUI extends Plugin {
 			const linkCommand: LinkCommand = editor.commands.get( 'link' )!;
 			const t = locale.t;
 
+			button.bind( 'isEnabled' ).to( linkCommand, 'value', href => !!href );
+
 			button.bind( 'href' ).to( linkCommand, 'value', href => {
 				return href && ensureSafeUrl( href, allowedProtocols );
 			} );
 
-			button.bind( 'label' ).to( linkCommand, 'value', href => {
+			button.bind( 'label' ).to( linkCommand, 'value', this, 'selectedLinksProviderLink', ( href, provider ) => {
```

Comment:
I'd avoid using multiple observables here. You can check the label on demand when the link value changes.